tests/test_program.py

```diff
@@ -127,8 +127,8 @@ def test_statements_no_params(self, program, name, wires):
     def test_observable_stmt(self, program, pref, wires):
         """Tests serializing an XIR program with observable statements."""
         xyz = "XYZ"
-        terms = [(xyz[i], w) for i, w in enumerate(wires)]
-        terms_str = " @ ".join(f"{t[0]}[{t[1]}]" for t in terms)
+        terms = [xir.ObservableFactor(xyz[i], None, w) for i, w in enumerate(wires)]
+        terms_str = " @ ".join(str(t) for t in terms)
```

Comment:
Might as well change `terms` to `factors` in the tests as well (including in several places below).
```suggestion
        factors = [xir.ObservableFactor(xyz[i], None, w) for i, w in enumerate(wires)]
        factors_str = " @ ".join(str(t) for t in factors)
```